Mathlib/CategoryTheory/Subobject/Basic.lean

```diff
@@ -0,0 +1,704 @@
+
+theorem le_of_comm {B : C} {X Y : Subobject B} (f : (X : C) ⟶ (Y : C)) (w : f ≫ Y.arrow = X.arrow) :
+    X ≤ Y := by
+  convert mk_le_mk_of_comm _ w <;> simp
+#align category_theory.subobject.le_of_comm CategoryTheory.Subobject.le_of_comm
+
+theorem le_mk_of_comm {B A : C} {X : Subobject B} {f : A ⟶ B} [Mono f] (g : (X : C) ⟶ A)
+    (w : g ≫ f = X.arrow) : X ≤ mk f :=
+  le_of_comm (g ≫ (underlyingIso f).inv) <| by simp [w]
+#align category_theory.subobject.le_mk_of_comm CategoryTheory.Subobject.le_mk_of_comm
+
+theorem mk_le_of_comm {B A : C} {X : Subobject B} {f : A ⟶ B} [Mono f] (g : A ⟶ (X : C))
+    (w : g ≫ X.arrow = f) : mk f ≤ X :=
+  le_of_comm ((underlyingIso f).hom ≫ g) <| by simp [w]
+#align category_theory.subobject.mk_le_of_comm CategoryTheory.Subobject.mk_le_of_comm
+
+/-- To show that two subobjects are equal, it suffices to exhibit an isomorphism commuting with
+    the arrows. -/
+@[ext]
+theorem eq_of_comm {B : C} {X Y : Subobject B} (f : (X : C) ≅ (Y : C))
+    (w : f.hom ≫ Y.arrow = X.arrow) : X = Y :=
+  le_antisymm (le_of_comm f.hom w) <| le_of_comm f.inv <| f.inv_comp_eq.2 w.symm
+#align category_theory.subobject.eq_of_comm CategoryTheory.Subobject.eq_of_comm
+
+-- porting note: removed @[ext]
+/-- To show that two subobjects are equal, it suffices to exhibit an isomorphism commuting with
+    the arrows. -/
+theorem eq_mk_of_comm {B A : C} {X : Subobject B} (f : A ⟶ B) [Mono f] (i : (X : C) ≅ A)
+    (w : i.hom ≫ f = X.arrow) : X = mk f :=
+  eq_of_comm (i.trans (underlyingIso f).symm) <| by simp [w]
+#align category_theory.subobject.eq_mk_of_comm CategoryTheory.Subobject.eq_mk_of_comm
+
+-- porting note: removed @[ext]
+/-- To show that two subobjects are equal, it suffices to exhibit an isomorphism commuting with
+    the arrows. -/
+theorem mk_eq_of_comm {B A : C} {X : Subobject B} (f : A ⟶ B) [Mono f] (i : A ≅ (X : C))
+    (w : i.hom ≫ X.arrow = f) : mk f = X :=
+  Eq.symm <| eq_mk_of_comm _ i.symm <| by rw [Iso.symm_hom, Iso.inv_comp_eq, w]
+#align category_theory.subobject.mk_eq_of_comm CategoryTheory.Subobject.mk_eq_of_comm
+
+-- porting note: removed @[ext]
+/-- To show that two subobjects are equal, it suffices to exhibit an isomorphism commuting with
+    the arrows. -/
+theorem mk_eq_mk_of_comm {B A₁ A₂ : C} (f : A₁ ⟶ B) (g : A₂ ⟶ B) [Mono f] [Mono g] (i : A₁ ≅ A₂)
+    (w : i.hom ≫ g = f) : mk f = mk g :=
+  eq_mk_of_comm _ ((underlyingIso f).trans i) <| by simp [w]
+#align category_theory.subobject.mk_eq_mk_of_comm CategoryTheory.Subobject.mk_eq_mk_of_comm
+
+-- We make `X` and `Y` explicit arguments here so that when `ofLe` appears in goal statements
+-- it is possible to see its source and target
+-- (`h` will just display as `_`, because it is in `Prop`).
+/-- An inequality of subobjects is witnessed by some morphism between the corresponding objects. -/
+def ofLe {B : C} (X Y : Subobject B) (h : X ≤ Y) : (X : C) ⟶ (Y : C) :=
```

Comment:
```suggestion
def ofLe {B : C} (X Y : Subobject B) (h : X ≤ Y) : (X : C) ⟶ (Y : C) :=
```
I think we have decided to keep `LE` capitalised even when it is part of a camelCase identifier, although this is fair from uniformly followed. I asked at https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Naming.20convention/near/350418574 for clarification on the consensus here.

I'll delegate this PR shortly, feel free to merge either with or without addressing this --- it's a widespread enough inconsistency there's no need to hold up this one in particular.

Reply:
Indeed, I was not sure. I shall change `Le` to `LE` later today.